### ISSUE TITLE
Search in Document tool creates invalid links if using TYPO3-Slugs

### ISSUE THREAD

**Issue by albig** [kitodo/kitodo-presentation#747]
## Description

With release 3.3 the links inside the textsnippets of the search in document tools are not valid if you have configured TYPO3 Slugs for the `tx_dlf[page]` parameter.

The error message is 

```
(1/1) #1518472189 TYPO3\CMS\Core\Error\Http\PageNotFoundException
The requested URL is not distinct
```

## Reproduction

Slug configuration in SiteConfiguration

```
routeEnhancers:
  KitodoWorkview:
    type: Plugin
    namespace: tx_dlf
    limitToPages:
      - 123 # (uid of workview)
    routePath: '/dlf/{id}/{page}'
    requirements:
      id: '(\d+)|(http.*xml)'
      page: \d+
```

Page of Workview with document uid=72043:

https://digital.slub-dresden.de/werkansicht/dlf/72043/1

Search inside the document, the hits are linked like this:

https://digital.slub-dresden.de/werkansicht/dlf/72043/1?tx_dlf[highlight_word]=Pieschen&tx_dlf[page]=3

TYPO3 now receives the tx_dlf[page] parameter twice: once by routing configuration (current page=1) and once by GET parameter (page=3). This is not accepted and an the PageNotFoundException exception is thrown.

See: https://github.com/TYPO3/typo3/blob/9.5/typo3/sysext/frontend/Classes/Middleware/PageResolver.php#L134

## Expected Behavior

The `baseUrl` for the text snippets in common Kitodo.Presentation usage must be taken from the action URL of the search form. This was the way [Kitodo.Presentation 3.2](https://github.com/kitodo/kitodo-presentation/blob/v3.2.2/Resources/Public/Javascript/PageView/SearchInDocument.js#L74) has been working.

In the example above, the `baseUrl` is https://digital.slub-dresden.de/werkansicht. All parameters have to be appended like the navigation plugin does. The resulting link must be

https://digital.slub-dresden.de/werkansicht?tx_dlf[id]=72043&tx_dlf[highlight_word]=Pieschen&tx_dlf[page]=3

I'm not sure about the impact to the DDB-Zeitungsportal.

## Environment

 - TYPO3 version: [e.g. 31]
 - Kitodo.Presentation 3.3.1
## Additional Context

This behaviour of TYPO3 won't change in TYPO3 10.4.

**Comment by beatrycze-volk** [comment]
The change was made exactly because the DDB-Zeitungsportal doesn't work in that way. There newspaper id is included in the URL and page is passed as a parameter. Including there id in parameter list was not working. We can assume that it was a special case and restore old solution (appending id, highlight word and page in parameters) or make some new more generic approach (including in configuration format in wich id, page and highlight word should be included in the link?).

I have written fast fix for our current problem (https://github.com/kitodo/kitodo-presentation/compare/master...beatrycze-volk:fix-link-search?expand=1).

**Comment by albig** [comment]
Closed by #748.
